File: solutions/c/triangle/2/triangle.c
#include "triangle.h"

static bool check_violation(triangle_t sides) {
    if ((sides.a + sides.b <= sides.c) || (sides.b + sides.c <= sides.a) || (sides.a + sides.c <= sides.b)) {
        return true;
    }
    return false;
}


bool is_equilateral(triangle_t sides) {
    if (check_violation(sides)) {
        return false;
    }
    if ((sides.a == sides.b) && (sides.b == sides.c)) {
        return true;
    }
    return false;
}

bool is_isosceles(triangle_t sides) {
    if (check_violation(sides)) {
        return false;
    }
    if ((sides.a == sides.b) || (sides.a == sides.c) || (sides.b == sides.c)){
        return true;
    }
    return false;
}

bool is_scalene(triangle_t sides) {
    if (check_violation(sides)) {
        return false;
    }
    if ((sides.a != sides.b) && (sides.b != sides.c)) {
        return true;
    }
    return false;
}

File: solutions/python/triangle/2/triangle.py
def equilateral(sides):
    if 0 in sides:
        return False

    a, b, c = sides

    return a == b == c


def isosceles(sides):

    if 0 in sides:
        return False

    a, b, c = sides

    # Triangle inequality
    if a + b < c or b + c < a or a + c < b:
        return False

    return a == b or b == c or a == c


def scalene(sides):
    if 0 in sides:
        return False
    a, b, c = sides

    # Triangle inequality
    if a + b < c or b + c < a or a + c < b:
        return False

    return a != b and b != c and a != c